src/backend/distributed/utils/ruleutils_95.c

```diff
@@ -3623,8 +3623,20 @@ get_variable(Var *var, int levelsup, bool istoplevel, deparse_context *context)
 	}
 	else
 	{
-		/* System column - name is fixed, get it from the catalog */
-		attname = get_rte_attribute_name(rte, attnum);
+		CitusRTEKind rtekind;
+
+		rtekind = GetRangeTblKind(rte);
+
+		if (rtekind == CITUS_RTE_SHARD || rtekind == CITUS_RTE_REMOTE_QUERY)
```

Comment:
Handling `CITUS_RTE_REMOTE_QUERY` is unnecessary; other `ruleutils_95` code only handles `CITUS_RTE_SHARD`. And it would be nice to keep the original code here unmodified, so could you instead handle the shard case like so?

```c
else if (GetRangeTblKind(rte) == CITUS_RTE_SHARD)
{
    /* System column on a Citus shared/remote relation */
    attname = get_relid_attribute_name(rte->relid, attnum);
}
else
{
    /* System column - name is fixed, get it from the catalog */
    attname = get_rte_attribute_name(rte, attnum);
}
```